shard/shard.go

```diff
@@ -0,0 +1,47 @@
+package shard
+
+import (
+	"crypto/md5"
+	"encoding/binary"
+	"math"
+
+	"github.com/grafana/dskit/yolo"
+)
+
+// Sharding Strategies.
+const (
+	ShardingStrategyDefault = "default"
+	ShardingStrategyShuffle = "shuffle-sharding"
+)
+
+var (
+	seedSeparator = []byte{0}
+)
+
+// ShuffleShardSeed returns seed for random number generator, computed from provided identifier.
+func ShuffleShardSeed(identifier, zone string) int64 {
+	// Use the identifier to compute an hash we'll use to seed the random.
```

Comment:
Could we fix this typo?
```suggestion
	// Use the identifier to compute a hash we'll use to seed the random.
```